src/NuGet.Clients/PackageManagement.VisualStudio/Migrate/ProjectJsonToPackageRefMigrator.cs

```diff
@@ -0,0 +1,135 @@
+            if (packageSpec.TargetFrameworks.Count > 1)
+            {
+                throw new InvalidOperationException(
+                    string.Format(
+                        CultureInfo.CurrentCulture,
+                        Strings.Error_MultipleFrameworks,
+                        project.MSBuildProjectPath));
+            }
+
+            var dependencies = new List<LibraryDependency>();
+            foreach (var targetFramework in packageSpec.TargetFrameworks)
+            {
+                dependencies.AddRange(targetFramework.Dependencies);
+            }
+
+            dependencies.AddRange(packageSpec.Dependencies);
+            foreach (var dependency in dependencies)
+            {
+                var includeFlags = dependency.IncludeType;
+                var privateAssetsFlag = dependency.SuppressParent;
+                var metadataElements = new List<string>();
+                var metadataValues = new List<string>();
+                if (includeFlags != LibraryIncludeFlags.All)
+                {
+                    metadataElements.Add("IncludeAssets");
+                    metadataValues.Add(LibraryIncludeFlagUtils.GetFlagString(includeFlags).Replace(',',';'));
+                }
+
+                if (privateAssetsFlag != LibraryIncludeFlagUtils.DefaultSuppressParent)
+                {
+                    metadataElements.Add("PrivateAssets");
+                    metadataValues.Add(LibraryIncludeFlagUtils.GetFlagString(privateAssetsFlag).Replace(',',';'));
+                }
+
+                await project.InstallPackageWithMetadataAsync(dependency.Name, dependency.LibraryRange.VersionRange, metadataElements, metadataValues);
+            }
+        }
+
+        private static void MigrateRuntimes(PackageSpec packageSpec, Microsoft.Build.Evaluation.Project buildProject)
+        {
+            var runtimes = packageSpec.RuntimeGraph.Runtimes;
+            var supports = packageSpec.RuntimeGraph.Supports;
+            var runtimeIdentifiers = new List<string>();
+            var runtimeSupports = new List<string>();
+            if (runtimes != null && runtimes.Count > 0)
+            {
+                runtimeIdentifiers.AddRange(runtimes.Keys);
+                
+            }
+
+            if(supports != null && supports.Count > 0)
+            {
+                runtimeSupports.AddRange(supports.Keys);
+            }
+
+            var union = string.Join(";", runtimeIdentifiers.Union(runtimeSupports));
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                buildProject.SetProperty("RuntimeIdentifiers", union);
+            });            
+        }
+
+        private static void RemoveProjectJsonReference(Microsoft.Build.Evaluation.Project buildProject, string projectJsonFilePath)
+        {
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
```

Comment:
Please remove JTF.Run from private methods and make them async. Leave JTF.Run for your public entrypoints.